test/unit/test_manifest_download.py

```diff
@@ -79,18 +88,43 @@ def test_manifest_download(self, download_func):
         self.assertEqual(download_func.call_count, len(self.manifest) - 1)
         self._assert_all_files_downloaded()
 
-    @patch('hca.dss.DSSClient._download_file', side_effect=_fake_download)
+    @patch('hca.dss.DSSClient._download_file', side_effect=_fake_download_file)
     def test_manifest_download_partial(self, _):
         _touch_file(self.dss._file_path(self.manifest[1][4]))
         self.dss.download_manifest_v2('manifest.tsv', 'aws')
         self._assert_all_files_downloaded()
 
     @patch('logging.Logger.warning')
     @patch('hca.dss.DSSClient._download_file', side_effect=[None, ValueError(), KeyError()])
-    def test_manifest_download_failed(self, download_func, warning_log):
+    def test_manifest_download_failed(self, _, warning_log):
         self.assertRaises(RuntimeError, self.dss.download_manifest_v2, 'manifest.tsv', 'aws')
         self.assertEqual(warning_log.call_count, 2)
 
+    @patch('hca.dss.DSSClient._get_file', side_effect=_fake_get_file)
+    def test_manifest_download_parallel(self, _):
+        """
+        The goal is to make sure the download of the file happens simultaneously in multiple threads.
+
+        For each file in the manifest always just "download" the same phony file.
+        """
+        old_download_file = self.dss._download_file
+        test_file = 'parallel_write_test_file'
+
+        def download_same_file(*_, **kwargs):
+            old_download_file('a uuid', 'fakeHASH', test_file, 'aws', 6)
```

Comment:
I'd explicitly state: `old_download_file('a uuid', 'fakeHASH', test_file, 'aws', size=6)` for clarity.

Reply:
I'll consider this, but I map end up refactoring this all together.

Reply:
Fair.  Then just a note here to please add it to the final version if necessary.